Block on mobile unit tests fail and instructions on how to debug

hypest opened the pull request:
## Description
This PR makes the native mobile tests Travis job necessary to succeed for the whole Travis run to "green".

## How has this been tested?
Online via Travis itself on this PR and locally by running `npm run test-unit:native:debug`.

## Types of changes
1. Lift the "allow to fail" flagging in the Travis config for the native mobile tests job
2. Add documentation on how to debug the native mobile unit tests
3. Fix the `test-unit:native:debug` npm script to indeed wait for the node debugger to attach

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->.

hypest committed:
Make native mobile tests a required-to-succeed job

hypest committed:
Proper Node debugger flag on native mobile tests

hypest committed:
Documentation on how to debug native mobile tests

gziolo commented:
Works for me, but I'm not the best person to validate changes, @Tug maybe? :)

Tug reviewed (approved):
Yep the `--inspect-brk` option should do the trick 👍
Nice addition to the documentation!

hypest committed:
Look in RN first so the `Platform` import picks up the correct one

Because there's also a Platform npm package installed too, running the
native mobile unit tests locally (at least on macOS) results in picking
up the wrong object. Putting the react-native/Libraries/Utilities first,
Jest manages to resolve the proper Platform import.

See
https://github.com/facebook/react-native/blob/v0.60.0/Libraries/react-native/react-native-implementation.js#L324-L326
for the import. The ambiguity seems fixed in RN v0.61.4 with
https://github.com/facebook/react-native/commit/62c605efe1a23923292f8ec4740bba8349cbcca2

hypest committed:
Merge branch 'master' into rnmobile/block-on-mobile-tests-fail

hypest committed:
Fix typo

hypest committed:
Add additional way to launch the Chrome dev tools for mobile

gziolo commented on test/native/jest.config.js:
```diff
@@ -39,7 +39,8 @@ module.exports = {
 	testURL: 'http://localhost/',
 	// Add the `Libraries/Utilities` subfolder to the module directories, otherwise haste/jest doesn't find Platform.js on Travis
-	moduleDirectories: [ 'node_modules', 'node_modules/react-native/Libraries/Utilities' ],
+	// And, add it first so https://github.com/facebook/react-native/blob/v0.60.0/Libraries/react-native/react-native-implementation.js#L324-L326 doesn't pick up the Platform npm module.
```
Super nit
```suggestion
	// and, add it first so https://github.com/facebook/react-native/blob/v0.60.0/Libraries/react-native/react-native-implementation.js#L324-L326
	// doesn't pick up the Platform npm module.
```

hypest replied:
Good catch, fixed with 91c96b7.

gziolo commented:
There is also the Testing Overview document:
https://github.com/WordPress/gutenberg/blob/master/docs/contributors/testing-overview.md

We should either move the details about testing there and cross-reference or the other way around. This is really important so it's easy to find details about the native app.

hypest committed:
Move the native mobile testing steps to testing-overview.md

hypest committed:
Fix typo

hypest commented:
> There is also the Testing Overview document:
https://github.com/WordPress/gutenberg/blob/master/docs/contributors/testing-overview.md
>
> We should either move the details about testing there and cross-reference or the other way around. This is really important so it's easy to find details about the native app.

Makes sense and I was thinking something along those line too 👍. Addressed with e5a0a97.

gziolo commented on docs/contributors/testing-overview.md:
```diff
@@ -386,3 +386,18 @@ To run unit tests only, without the linter, use `npm run test-unit-php` instead.
 [snapshot testing]: https://jestjs.io/docs/en/snapshot-testing.html
 [update snapshots]: https://jestjs.io/docs/en/snapshot-testing.html#updating-snapshots
+
+## Native mobile testing
```
Can you move it up just before the section about E2E testing? It's better when it's grouped together with the regular unit testing, it's nearly identical after all :)

hypest replied:
Sure, addressed with 3064350.

gziolo reviewed (approved):
Thank you for working for all those additions to contributors guide ❤️

hypest committed:
Move the instructions closer to the regular unit testing sections

gziolo commented:
Nice, thanks, feel free to merge 👍

hypest commented:
I'm going to leave this PR open for another day before merging. We brought this up more widely in the #core-editor weekly chat yesterday so, will leave some time for people to comment if any.

hypest commented:
No new comments here so, will go ahead and merge.
